src/foam/nanos/analytics/DAOFoldManager.js

```diff
@@ -85,8 +85,6 @@ setCloseTimeExpr(new foam.glang.EndOfTimeSpan.Builder(getX())
             c.setCloseTime(id.getCloseTime());
             c.setKey(key);
             c.init_();
-          } else {
-            c = (Candlestick) c.fclone();
```

Comment:
If you aren't going to clone the object, then I don't think you need to put it back in the DAO either. Just move the getDao().put(c) line to the last line of the 'if' block above (line 87.5).